### PR TITLE
Templates API V4 - Allow configuring template admins via the API.

### PR DESCRIPTION
**Pull request by varshavaradarajan**

### CODE REVIEW

File: server/webapp/WEB-INF/rails.new/app/controllers/api_v4/admin/templates_controller.rb [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+##########################################################################
+
+module ApiV4
+  module Admin
+    class TemplatesController < ApiV4::BaseController
+      before_action :check_admin_user_and_401
```
Comment (ketan) [inline server/webapp/WEB-INF/rails.new/app/controllers/api_v4/admin/templates_controller.rb]: template admin?

File: server/webapp/WEB-INF/rails.new/app/controllers/api_v4/admin/templates_controller.rb [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+      end
+
+      def load_template(template_name = params[:template_name])
+        result = HttpLocalizedOperationResult.new
+        @template = template_config_service.loadForView(template_name, result)
+        raise RecordNotFound if @template.nil?
+      end
+
+      def check_for_stale_request
+        if request.env["HTTP_IF_MATCH"] != "\"#{Digest::MD5.hexdigest(get_etag_for_template(@template))}\""
```
Comment (ketan) [inline server/webapp/WEB-INF/rails.new/app/controllers/api_v4/admin/templates_controller.rb]: Use `%Q{}` so as to not have to escape stuff, and improve readability.

File: server/webapp/WEB-INF/rails.new/app/controllers/api_v4/admin/templates_controller.rb [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+      def destroy
+        result = HttpLocalizedOperationResult.new
+        template_config_service.deleteTemplateConfig(current_user, @template, result)
+        render_http_operation_result(result)
+      end
+
+      def load_template(template_name = params[:template_name])
+        result = HttpLocalizedOperationResult.new
+        @template = template_config_service.loadForView(template_name, result)
+        raise RecordNotFound if @template.nil?
```
Comment (ketan) [inline server/webapp/WEB-INF/rails.new/app/controllers/api_v4/admin/templates_controller.rb]: `unless @template`?

File: server/webapp/WEB-INF/rails.new/app/presenters/api_v4/config/pipeline_config_summary_representer.rb [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+# limitations under the License.
+##########################################################################
+
+module ApiV4
+  module Config
+    class PipelineConfigSummaryRepresenter < ApiV4::BaseRepresenter
+      alias_method :pipeline, :represented
+
+      link :self do |opts|
+        opts[:url_builder].apiv3_admin_pipeline_url(pipeline.getName())
```
Comment (ketan) [inline server/webapp/WEB-INF/rails.new/app/presenters/api_v4/config/pipeline_config_summary_representer.rb]: `apiv3` -> `apiv4` in a bunch of places.

File: server/webapp/WEB-INF/rails.new/config/routes.rb [hunk 5]
```diff
@@ -311,12 +311,15 @@
...
         patch :update, on: :member
       end
-
       # for some reasons using the constraints breaks route specs for routes that don't use constraints, so an ugly hax
       get 'agents', action: :index, controller: 'agents'
       patch 'agents', action: :bulk_update, controller: 'agents'
       delete 'agents', action: :bulk_destroy, controller: 'agents'
 
+      namespace :admin do
+        resources :templates, param: :template_name, only: [:index, :show, :update, :create, :destroy], constraints: {template_name: TEMPLATE_NAME_FORMAT}
```
Comment (ketan) [inline server/webapp/WEB-INF/rails.new/config/routes.rb]: `only: [:index, :show, :update, :create, :destroy]`, are there others?

File: server/webapp/WEB-INF/rails.new/app/controllers/api_v4/admin/templates_controller.rb (hunk 1 above)
Comment (varshavaradarajan) [inline server/webapp/WEB-INF/rails.new/app/controllers/api_v4/admin/templates_controller.rb]: Part of another PR.

File: server/webapp/WEB-INF/rails.new/app/presenters/api_v4/config/pipeline_config_summary_representer.rb (hunk 4 above)
Comment (varshavaradarajan) [inline server/webapp/WEB-INF/rails.new/app/presenters/api_v4/config/pipeline_config_summary_representer.rb]: apiv3 for pipelines. pipelines doesn't have apiv4.

File: server/webapp/WEB-INF/rails.new/config/routes.rb (hunk 5 above)
Comment (varshavaradarajan) [inline server/webapp/WEB-INF/rails.new/config/routes.rb]: Nope. These are the only ones for configuration of templates.

**Pull request merged**